pyteal/ast/ecdsa_test.py

```diff
@@ -0,0 +1,197 @@
+    assert expr.type_of() == pt.TealType.uint64
+
+    expected = pt.TealSimpleBlock(
+        [
+            pt.TealOp(compressed_pubkey, pt.Op.byte, '"XY"'),
+            pt.TealOp(pubkey, pt.Op.ecdsa_pk_decompress, pt.EcdsaCurve.Secp256k1),
+            pt.TealOp(
+                pubkey.output_slots[1].store(), pt.Op.store, pubkey.output_slots[1]
+            ),
+            pt.TealOp(
+                pubkey.output_slots[0].store(), pt.Op.store, pubkey.output_slots[0]
+            ),
+            pt.TealOp(args[0], pt.Op.byte, '"data"'),
+            pt.TealOp(args[1], pt.Op.byte, '"sigA"'),
+            pt.TealOp(args[2], pt.Op.byte, '"sigB"'),
+            pt.TealOp(
+                pubkey.output_slots[0].load(), pt.Op.load, pubkey.output_slots[0]
+            ),
+            pt.TealOp(
+                pubkey.output_slots[1].load(), pt.Op.load, pubkey.output_slots[1]
+            ),
+            pt.TealOp(expr, pt.Op.ecdsa_verify, pt.EcdsaCurve.Secp256k1),
+        ]
+    )
+
+    actual, _ = expr.__teal__(teal5Options)
+    actual.addIncoming()
+    actual = pt.TealBlock.NormalizeBlocks(actual)
+
+    with pt.TealComponent.Context.ignoreExprEquality():
+        assert actual == expected
+
+
+def test_ecdsa_verify_recovered_pk():
+    args = [pt.Bytes("data"), pt.Int(1), pt.Bytes("sigA"), pt.Bytes("sigB")]
+    pubkey = pt.EcdsaRecover(
+        pt.EcdsaCurve.Secp256k1, args[0], args[1], args[2], args[3]
+    )
+    expr = pt.EcdsaVerify(pt.EcdsaCurve.Secp256k1, args[0], args[2], args[3], pubkey)
+    assert expr.type_of() == pt.TealType.uint64
+
+    expected = pt.TealSimpleBlock(
+        [
+            pt.TealOp(args[0], pt.Op.byte, '"data"'),
+            pt.TealOp(args[1], pt.Op.int, 1),
+            pt.TealOp(args[2], pt.Op.byte, '"sigA"'),
+            pt.TealOp(args[3], pt.Op.byte, '"sigB"'),
+            pt.TealOp(pubkey, pt.Op.ecdsa_pk_recover, pt.EcdsaCurve.Secp256k1),
+            pt.TealOp(
+                pubkey.output_slots[1].store(), pt.Op.store, pubkey.output_slots[1]
+            ),
+            pt.TealOp(
+                pubkey.output_slots[0].store(), pt.Op.store, pubkey.output_slots[0]
+            ),
+            pt.TealOp(args[0], pt.Op.byte, '"data"'),
+            pt.TealOp(args[1], pt.Op.byte, '"sigA"'),
+            pt.TealOp(args[2], pt.Op.byte, '"sigB"'),
+            pt.TealOp(
+                pubkey.output_slots[0].load(), pt.Op.load, pubkey.output_slots[0]
+            ),
+            pt.TealOp(
+                pubkey.output_slots[1].load(), pt.Op.load, pubkey.output_slots[1]
+            ),
+            pt.TealOp(expr, pt.Op.ecdsa_verify, pt.EcdsaCurve.Secp256k1),
+        ]
+    )
+
+    actual, _ = expr.__teal__(teal5Options)
+    actual.addIncoming()
+    actual = pt.TealBlock.NormalizeBlocks(actual)
+
+    with pt.TealComponent.Context.ignoreExprEquality():
+        assert actual == expected
+
+
+def test_ecdsa_invalid():
```

Comment:
I'd also try compiling with TEAL 4 to check that it raises a `TealInputError` as a sanity check.

I'm also personally curious what would happen if the first argument was something other than the `EcdsaCurve` enum, e.g. what happens if we pass in `2`?